libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
Your component must be decoupled from the application it is running in. The only thing it should know about is the Container. This way your model, for example, can work with the front-end application, the back-end application or even a custom JApplicationWeb or a CLI application without any change in your code base. We currently have JFactory to do that but it's an ugly static interface with several limitations (testability is right at the top of the list but NOT the only one). A container is a more elegant, powerful and interchangeable solution.